lib/component-data/pubsub.js

```diff
@@ -132,6 +148,68 @@ function shouldSubscribe(name, eventID) {
   return !_.includes(callStacks[eventID], name);
 }
 
+/**
+ * set payload values for property to publish
+ * @param {object} prop
+ * @param {object} payload  to modify
+ * @param {*} value    to publish
+ * @param {string} eventID
+ * @param {string} uri
+ */
+function setPayload(prop, payload, value, eventID, uri) { // eslint-disable-line
+  const propName = prop.name,
+    propScope = prop.scope;
+
+  if (hasSubscribers(propName)) {
+    // there are subscribers to this property!
+    // see if they should subscribe (e.g. they haven't already published) and construct
+    // a payload of the data that needs to be sent to these components
+    _.forOwn(subscriberRegistery[propName], (subscribedFields, subscriberName) => {
+      if (shouldSubscribe(subscriberName, eventID)) {
+        payload[subscriberName] = {
+          data: _.reduce(Array.from(subscribedFields), (subData, subscribedField) => _.assign(subData, { [subscribedField]: value }), _.get(payload, `${subscriberName}.data`, {})),
+          scope: propScope,
+          publisherURI: uri
+        };
+      }
+    });
+  }
+}
+
+/**
+ * quickly determine if a component contains another component
+ * by looking at the dom
+ * @param  {string}  parent uri
+ * @param  {string}  child  uri
+ * @return {Boolean}
+ */
+function isComponentInside(parent, child) {
+  return !!find(`[${refAttr}="${parent}"] [${refAttr}="${child}"]`);
+}
+
+/**
+ * determine if a subscriber fits the publisher's scope
+ * note: allowed scopes are currently 'children', 'parent', and 'global'
+ * @param  {string} scope
+ * @param  {string} publisherURI
+ * @param  {string} subscriberURI
+ * @return {boolean}
+ */
+function subscriberInScope(scope, publisherURI, subscriberURI) {
+  if (scope === 'global') {
+    // 'global' scope is the default
+    return true;
+  } else if (_.includes(['children', 'child'], scope)) {
```

Comment:
why is it checking for both 'children' and 'child', 'parents' and 'parent'?

Reply:
While `children` and `parents` is the documented, acceptable option, I didn't want to penalize people who might mistakenly type `child` or `parent` (since they're very easy to confuse _and_ are frequently used to describe this kind of scoping in other systems / frameworks)